src/cephadm/cephadm.py

```diff
@@ -2961,107 +2976,13 @@ def get_ceph_volume_container(ctx: CephadmContext,
-        mounts = get_container_mounts(ctx, ident)
-    elif daemon_type == CephExporter.daemon_type:
-        ceph_exporter = CephExporter.create(ctx, ident)
-        entrypoint = ceph_exporter.default_entrypoint()
-        ceph_exporter.customize_container_envs(ctx, envs)
-        ceph_exporter.customize_container_args(ctx, container_args)
-        ceph_exporter.customize_process_args(ctx, d_args)
-        mounts = get_container_mounts(ctx, ident)
-    elif daemon_type == HAproxy.daemon_type:
-        haproxy = HAproxy.create(ctx, ident)
-        haproxy.customize_container_args(ctx, container_args)
-        haproxy.customize_process_args(ctx, d_args)
-        mounts = get_container_mounts(ctx, ident)
-    elif daemon_type == Keepalived.daemon_type:
-        keepalived = Keepalived.create(ctx, ident)
-        keepalived.customize_container_envs(ctx, envs)
-        keepalived.customize_container_args(ctx, container_args)
-        mounts = get_container_mounts(ctx, ident)
-    elif daemon_type == CephNvmeof.daemon_type:
-        nvmeof = CephNvmeof.create(ctx, ident)
-        nvmeof.customize_container_args(ctx, container_args)
-        binds = get_container_binds(ctx, ident)
-        mounts = get_container_mounts(ctx, ident)
-    elif daemon_type == CephIscsi.daemon_type:
-        iscsi = CephIscsi.create(ctx, ident)
-        entrypoint = iscsi.default_entrypoint()
-        iscsi.customize_container_args(ctx, container_args)
-        # So the container can modprobe iscsi_target_mod and have write perms
-        # to configfs we need to make this a privileged container.
-        privileged = True
-        binds = get_container_binds(ctx, ident)
-        mounts = get_container_mounts(ctx, ident)
-    elif daemon_type == CustomContainer.daemon_type:
-        cc = CustomContainer.init(ctx, ident.fsid, ident.daemon_id)
-        entrypoint = cc.default_entrypoint()
-        host_network = False
-        cc.customize_container_envs(ctx, envs)
-        cc.customize_container_args(ctx, container_args)
-        cc.customize_process_args(ctx, d_args)
-        binds = get_container_binds(ctx, ident)
-        mounts = get_container_mounts(ctx, ident)
-    elif daemon_type == SNMPGateway.daemon_type:
-        sg = SNMPGateway.create(ctx, ident)
-        sg.customize_container_args(ctx, container_args)
-        sg.customize_process_args(ctx, d_args)
-
-    _update_container_args_for_podman(ctx, ident, container_args)
-    return CephContainer.for_daemon(
-        ctx,
-        ident=ident,
-        entrypoint=entrypoint,
-        args=d_args,
-        container_args=container_args,
-        volume_mounts=mounts,
-        bind_mounts=binds,
-        envs=envs,
-        privileged=privileged,
-        ptrace=ptrace,
-        host_network=host_network,
+    daemon = daemon_form_create(ctx, ident)
+    assert isinstance(daemon, ContainerDaemonForm)
+    privileged = ident.daemon_type in {'mon', 'osd', CephIscsi.daemon_type}
```

Comment:
NIT, but I think it would be nice to standardize the list of daemon types that need to run with `--privileged`. There's another spot in this PR that says
```
privileged = self.identity.daemon_type in ['mon', 'osd']
```
which already doesn't include the ISCSI daemon type as this set does here. If we had one constant somewhere that listed all the types that need it,  we could just reference it whenever needed.

Reply:
Good point. I generally feel that this knowledge should be kept within each ContainerDaemonForm class but I was already running out of energy to "refactor all the things".

Reply:
That said, I think this PR is OK as is as these kinds of improvements can be done later.